### ISSUE TITLE
Embedly Losing Extract API

### ISSUE THREAD

**Issue by arist0tl3** [VulcanJS/Vulcan#1588]
Got an email from a client today and saw that Embedly is moving to a new pricing structure, with access to the Extract API coming in at $99/mo. Not really a feasible jump from $9/mo.

Looks like the Cards could be hacked into thumbnails with some css to hide almost everything except the image, but this seems like a clunky alternative to the current implementation.

[Noembed](https://noembed.com/) provides an alternative for `getEmbedlyData`, but there's still the issue of having a convenient way to display cached images from a CDN that are cropped. [Imgix](https://imgix.com) has been great in my experience, but that's another layer of complexity.

Just curious if anyone else has thoughts?

**Comment by SachaG** [comment]
Is that for displaying images off of Embedly's CDN, or even for getting the metadata in the first place? Because for the displaying part, you can use the `nova:cloudinary` package to cache images on Cloudinary so you don't pay a big Embedly fee. 

If getting the metadata in the first place is not going to be free anymore then I guess we need a solution. Maybe Noembed+Cloudinary would work?

(BTW I also checked out Imgix but at the time they didn't support uploading an image straight from a URL, you had to upload the actual image file through your own servers first, which didn't fit our use case)

**Comment by arist0tl3** [comment]
Even getting the metadata is restricted. 

There are two tiers moving forward from April 1st - Cards, which basically transforms a filler div into a card with all of the relevant pieces (thumbnail, link, author, etc). and API, which is a bundle of all of their APIs (including extract). Seems like if you want to keep using extract, then you need the 99/mo plan.

Cards obviously uses an extraction method to generate the resulting card, as the only piece of data that you provide in the filler div is the url. A quick look at the [platform.js](http://cdn.embedly.com/widgets/platform.js) client code that is required for the Cards shows them using the extract API/methods there. But I have to assume that they're going to want to prevent API access if you aren't paying for it...

Also, you are correct that Imgix requires a source, so that would be problematic. Still, when I see that 99/mo price tag, it makes me start thinking outside the box, hah.

**Comment by dominictracey** [comment]
@ucefkh you mentioned you have an embedly package? Does it provide a workaround for the $99 problem? :1st_place_medal:

**Comment by meteorplus** [comment]
@dominictracey yes my package costs $99 one time payment and you are totally independent from embedly once and for all

**Comment by dominictracey** [comment]
An opportunist - I like it!

**Comment by arist0tl3** [comment]
@meteorplus Can you provide some info on the package?

**Comment by meteorplus** [comment]
@dominictracey  :D thanks

@arist0tl3  does everything as embedly the only thing you will have to add your s3 api key or cloudinary (both upload from the browser) if interested pm me on slack will get back to you asap!

**Comment by dominictracey** [comment]
Embedly provides all the "real time" scraping on their servers with back channel sending of the meta data to the client. Providing a component that does meta-data extraction on the client is most certainly not the same thing and has a variety of drawbacks. An option I've looked at is the venerable YQL (https://developer.yahoo.com/yql/).

**Comment by manriquef** [comment]
@meteorplus I am interested in the package you have, how do I contact you? is your slack name the same?

**Comment by meteorplus** [comment]
@dominictracey I never said extraction would be on the client only uploading!

@manriquef contact ucefkh on slack ;)

**Comment by townmulti** [comment]
Embedly is also offering a free plan with embedly cards but will have embedly branding on the cards.  Though I'm experiencing a scrolling issue with them in my app. Anybody else using the embedly cards with react? `dangerouslySetInnerHTML` is not working either.

**Comment by arist0tl3** [comment]
Just a heads up, I was able to hack on the embedly package a bit and have iframely pulling the metadata successfully. Caveats:

1. Iframely's free "developer" tier gets you 1k hits per month, and 10k is the next tier for $29/mo
2. The thumbnail is pulled right from the source with no resizing (I just used some css to fix up the thumbs) so no CDN is built into this solution (and it's technically hotlinking)
3. Aspect ratios are maintained, so if you are used to 300x233 (or a similar fixed ratio), then this may not be a solution

That being said, I'm happy to put together a gist with the changes if anyone is interested. I can try a PR, but we are running Telescope legacy and I have honestly fallen out of touch with this codebase, so not sure I'm the guy for that job.

**Comment by meteorplus** [comment]
For people who wants a good affordable alternative to Embedly today and don't want to change any code, you can get the same service at low price $99 per year instead of paying $99 per month on our platform: EmbedAPI.com

and here is our package ready to use with the latest version of vulcan
https://github.com/meteorplus/embedapi

Enjoy!
